assets/js/phoenix_live_view.js

```diff
@@ -2379,8 +2379,8 @@ export class View {
 
   destroyHook(hook){
     hook.__trigger__("destroyed")
-    hook.__cleanup__()
     delete this.viewHooks[ViewHook.elementID(hook.el)]
+    hook.__cleanup__()
```

Comment:
Note: the call to `ViewHook.elementID(hook.el)` uses the `phxHookId`, so we need to execute that first and cleanup afterwards.